### PR TITLE
Update copy on old modal for personal/premium upsell features

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/red-team/issues/309

## Proposed Changes

Currently, the upsell modal has a feature list based around the upsell only ever upselling to the premium plan. Since we now handle upselling the the personal plan this modal needs updating to include features specific to the personal plan.

Before - only one feature list, shown for both premium and personal plan upsells

Screenshot(165)

After - specific feature lists for personal and premium

Screenshot 2024-12-11 at 16-56-05 Jetpack Stats ‹ Site test — WordPress com

Screenshot 2024-12-11 at 16-55-38 Jetpack Stats ‹ Site test — WordPress com

## Testing Instructions

The personal upsell modal is only shown when the feature flag enabled, without it, it only shows premium plans.

* View stats for a free site, click the upsell in the date picker, it should show the new feature list for the premium plans

### CODE REVIEW

File: client/my-sites/stats/stats-upsell-modal/index.tsx [hunk 1]
```diff
@@ -133,44 +134,94 @@ export default function StatsUpsellModal( { siteId }: { siteId: number } ) {
...
+					{ translate( '14-day money-back guarantee' ) }
+				</div>
+			</div>
+			<div className="stats-upsell-modal__feature">
+				<Gridicon icon="checkmark" size={ 18 } />
+				<div className="stats-upsell-modal__feature-text">
+					{ translate( 'All %(planName)s plan features', {
+						args: { planName: plan?.productNameShort ?? '' },
+					} ) }
+				</div>
```
Comment (lsl) [inline client/my-sites/stats/stats-upsell-modal/index.tsx:223]: I made some educated guesses about the copy here but after posting the PR and testing I think we should align this copy with the in-page modal.

**Review by kangzj — APPROVED** [review]
LGTM 🚀

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17059011

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17059011).

**Pull request merged**